src/main/java/com/cognifide/aemrules/checks/slingquery/SlingQueryImplicitStrategyCheck.java

```diff
@@ -0,0 +1,132 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.aemrules.checks.slingquery;
+
+import com.cognifide.aemrules.tag.Tags;
+import java.util.HashMap;
+import java.util.Map;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.plugins.java.api.JavaFileScanner;
+import org.sonar.plugins.java.api.JavaFileScannerContext;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.ExpressionStatementTree;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.VariableTree;
+
+@Rule(
+    key = SlingQueryImplicitStrategyCheck.RULE_KEY,
+    name = SlingQueryImplicitStrategyCheck.RULE_MESSAGE,
+    priority = Priority.MINOR,
+    tags = Tags.AEM
+)
+public class SlingQueryImplicitStrategyCheck extends BaseTreeVisitor implements JavaFileScanner {
+
+    protected static final String RULE_KEY = "AEM-19";
+
+    protected static final String RULE_MESSAGE = "Implicit search strategy used in Sling Query";
+
+    private static final String FIND_METHOD = "find";
+
+    private static final String SEARCH_STRATEGY_METHOD = "searchStrategy";
+
+    private static final String SLING_QUERY = "SlingQuery";
+
+    private String slingQueryName = null;
+
+    private Map<String, SlingQueryStates> slingQueries = new HashMap();
+
+    private JavaFileScannerContext context;
+
+    private enum SlingQueryStates {
+        NOT_USED,
+        FIND_USED_WITHOUT_STRATEGY,
+        STRATEGY_USED,
+        ISSUE_RETURNED
+    }
+
+    @Override
+    public void scanFile(JavaFileScannerContext context) {
+        this.context = context;
+        scan(context.getTree());
+    }
+
+    @Override
+    public void visitVariable(VariableTree tree) {
+        String variableName = tree.simpleName().name();
+        if (isSlingQuery(tree)) {
+            slingQueries.put(variableName, SlingQueryStates.NOT_USED);
+            slingQueryName = variableName;
+        }
+        super.visitVariable(tree);
+        // This part of code will trigger directly after initialization
+        if (findWithoutStrategyWasUsedOnSlingQuery()) {
+            context.reportIssue(this, tree, RULE_MESSAGE);
+            slingQueries.put(slingQueryName, SlingQueryStates.ISSUE_RETURNED);
+        }
+    }
+
+    @Override
+    public void visitMethodInvocation(MethodInvocationTree tree) {
+        if (isFindMethod(tree) && isThisANewSlingQuery()) {
+            slingQueries.replace(slingQueryName, SlingQueryStates.FIND_USED_WITHOUT_STRATEGY);
+        } else if (isSearchStrategyMethod(tree)) {
+            slingQueries.replace(slingQueryName, SlingQueryStates.STRATEGY_USED);
+        }
+        super.visitMethodInvocation(tree);
+    }
+
+    @Override
+    public void visitExpressionStatement(ExpressionStatementTree tree) {
+        String variableName = tree.firstToken().text();
```

Comment:
Why sow general name. You can start from slingQueryName = tree.firstToken().text()

Reply:
done